### DISCUSSION TITLE
OsmAnd format for POI

### DISCUSSION THREAD

**Discussion by ZeljkoBG** [osmandapp/OsmAnd#19534] [Ideas]
I suggest that OsmAnd have its own format for creating or importing custom POI, to leave GPX.
A new format to have more options. If you can add your own icons...

**Comment by pebogufi** [comment]
NO !!!
The gpx format is interchangeable, this is very big advantage !
It is better to add new tags, for additional features if they are not commonly known.

**Comment by ZeljkoBG** [comment]
The GPX format is 100 years old and contains nothing but coordinates. That's why many (one man GPS) use it because it's simple.
Navigations have either their own format or some 'more complex'.

**Reply by mariush444** [reply]
You can use any icon that you can see in vector obf map (not only osmand icon)
It seems to be enough.

**Reply by ZeljkoBG** [reply]
Of course not. Others have said it's not enough.
There are plenty of websites on the internet that give free POI points that are not available on OSM, HERE, TT,... maps. And which contains icons of these companies. It is not the same to see a recognizable icon or OsmAnd 'star'.

**Reply by mariush444** [reply]
Sure. But changing gpx format it is not good idea either. Because it will be not gpx then so exchanging files between devices and system will be lost. (I vote against)
Some adding tags with link to icon probably is possible.

Additionaly I can imagine 2 solutions.
1. KMZ supported nativly as gpx in osmand. Then you can use your own icons. But probably there will be problem with performance.
2. Favorities are in DB so it is posibble to store own icon there or store a link to icon in lets say icon folder.

PS
Mention websites by you supports their own icons and not anyothers as I tested sometimes ago.

PS2
What you say about star is a different problem and it is about reasonable converter.
Currently there is converter for KMZ and garmin as I remember. So there is not only a star but every OSM icon and every color.

**Reply by mariush444** [reply]
additional info

taging - extensions in gpx
https://osmand.net/docs/technical/osmand-file-formats/osmand-gpx
https://www.osmand.net/docs/user/map/track-context-menu#display-custom-gpx-tags

converters to kmz to "osmand" gpx with icons
https://github.com/mariush444/gmapIcons2osmand
https://github.com/tmusolf/KMLtoOSMAndGPX

converter from garmin
https://github.com/maurizioandreotti/GPX-Basecamp-2-OsmAnd

as icons in osmand you can use about 1000 different icons. list in pdf.
[osmand - icons.pdf](https://github.com/osmandapp/OsmAnd/files/14966976/osmand.-.icons.pdf)

**Reply by ZeljkoBG** [reply]
I looked it up on github. Every option, in my opinion, is incomplete.

I wasn't clear. The GPX format doesn't matter either. My proposal is something had TomTom, Primo, Sygic 

**Example**  - If there is an option import 
and
Avia petrol.gpx/kmz/.../
and
Avia petrol.bmp

OR

Small Program (*.exe or *.jar) which will convert both formats to the third, OsmAnd format

Anyway, anything

**Reply by vshcherb** [reply]
Nope you can't import custom icons, you can just do it in a program select some icons and export it. Open with text file and see the format, that's what supported and that's the easiest

**Reply by ZeljkoBG** [reply]
I know where I am. I know he can't. This is part of the discussion "Ideas", where I proposed my idea, **"that it can"**.

**Reply by ZeljkoBG** [reply]
Perhaps as well iGO
Apartmani u Grckoj.kml
Apartmani u Grckoj.bmp
Osman used them as one of the **Apartmani u Grckoj.ZIP** 

KML
```
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<kml xmlns="http://earth.google.com/kml/2.2">
<Document><name>Apartmani u Grckoj</name><open>1</open><metadata><igoicon><filename>Grcka.bmp</filename></igoicon></metadata>
<Folder><name>Nea Skioni</name><metadata><igoicon><filename>Apartmani u Grckoj.bmp</filename></igoicon></metadata>
<Placemark><name>Vila Katerina</name><Point><coordinates>23.52727,39.94832,0</coordinates></Point></Placemark>
</Folder>
<Folder><name>Olympic Beach</name><metadata><igoicon><filename>Apartmani u Grckoj.bmp</filename></igoicon></metadata>
<Placemark><name>Vila Konstantin</name><Point><coordinates>22.58675,40.23976,0</coordinates></Point></Placemark>
</Folder>
</Document>
</kml>
```

**Reply by 0709wiwiwi** [reply]
@ZeljkoBG 
An example of a kmz track.  Shows icons and photo.  Import into GE or Locus map.
[BOM_KMZ.kmz.txt](https://github.com/user-attachments/files/15759922/BOM_KMZ.kmz.txt)

**Reply by ZeljkoBG** [reply]
Sorry, I need clarification on BOM_KMZ.kmz
1. You propose this as an idea
or
2. Your BOM_KMZ.kmz file works

**Reply by 0709wiwiwi** [reply]
2. It works by the GE known standard.

**Reply by ZeljkoBG** [reply]
How to add BOM_KMZ.kmz to OsmAnd? I miss that information.

**Reply by 0709wiwiwi** [reply]
OsmAnd does not support !  Test it in the apps that accept the format.
